src/harness/unittests/tsserverProjectSystem.ts

```diff
@@ -1539,6 +1539,45 @@ namespace ts.projectSystem {
             }
         });
 
+        it("removes version numbers correctly", () => {
+            const testData: [string, string][] = [
+                ["jquery-max", "jquery-max"],
+                ["jquery.min", "jquery"],
+                ["jquery-min.4.2.3", "jquery"],
+                // ["jquery.4.2-test.js", "jquery"],
+                ["jquery.min.4.2.1", "jquery"],
+                // ["jquery.7.min.js", "jquery"],

```

Comment:
This is pretty much the same format as the case you thought didn't work. Any reason not to include it? (without the .js, if that's already removed by the time it gets here).

Also, why leave commented out tests? It looks like they should pass but are currently failing. We should just remove them if that is not the behavior we want (or enable them and change the expectations to the correct behavior).